adobe/spectrum-web-components
# fix(reactive-controllers): update focusable element's tab-index to 0 on accepting focus

## Description

A focusable element should set its tab-index=0 when receiving the focus (if it is being managed by a focusGroupController).
Earlier the controller used to set the tabindex to 0 when the focus was received using a keyboard interaction. With this PR the tabindex would be set to 0 on mouse click also.

## Related issue(s)



- #4649

## Motivation and context

Accessibility

## How has this been tested?




-   [x] sp-action-group
    1. Go [here](https://fix-tab-index-management--spectrum-web-components.netlify.app/storybook/?path=/story/action-group--has-action-menu-as-child)
    2. Open Inspect elements tab.
    3. Inspect any child of the action-group.
    4. Click on that same child and see the tabindex attribute to be 0 for that child in the inspector.
    
-   [x] sp-sidenav
    1. Go [here](https://fix-tab-index-management--spectrum-web-components.netlify.app/storybook/?path=/story/sidenav--manage-tab-index)
    2. Open Inspect elements tab.
    3. Inspect any child of the sidenav that is not selected.
    4. Click on that same child and see the tabindex attribute to be 0 for that child in the inspector.
        
-   [x] sp-radio
    1. Go [here](https://fix-tab-index-management--spectrum-web-components.netlify.app/storybook/?path=/story/radio--horizontal-group)
    2. Open Inspect elements tab.
    3. Inspect any child of the radiogroup that is not selected.
    4. Click on that same child and see the tabindex attribute to be 0 for that child in the inspector.
    
-   [x] sp-tabs
    1. Go [here](https://fix-tab-index-management--spectrum-web-components.netlify.app/storybook/?path=/story/tabs--default)
    2. Open Inspect elements tab.
    3. Inspect any child of the tabs that is not selected.
    4. Click on that same child and see the tabindex attribute to be 0 for that child in the inspector.
        
-   [x] sp-grid
    1. Go [here](https://fix-tab-index-management--spectrum-web-components.netlify.app/storybook/?path=/story/grid--default)
    2. Open Inspect elements tab.
    3. Inspect any child of the grid that is not selected.
    4. Click on that same child and see the tabindex attribute to be 0 for that child in the inspector.

    

## Screenshots (if appropriate)

## Types of changes




## Checklist




## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.
